feature-libs/cart/added-to-cart-toast/components/added-to-cart-toast/added-to-cart-toast.component.ts

```diff
@@ -0,0 +1,185 @@
+
+  constructor(
+    protected winRef: WindowRef,
+    protected addedToCartToastConfig: AddedToCartToastConfig,
+    protected addedToCartToastService: AddedToCartToastService,
+    protected eventService: EventService,
+    protected productService: ProductService,
+    protected renderer: Renderer2,
+    protected routerService: RoutingService
+  ) {}
+
+  ngOnInit() {
+    this.headerElement = this.winRef.document.querySelector('header');
+
+    if (!this.customClass) this.customClass = 'cx-added-cart-toast';
+
+    this.baseClass = `${this.customClass}`;
+    this.toastContainerClass = this.toastContainerBaseClass;
+
+    if (this.addedToCartToastConfig.addedToCartToast?.timeout)
+      this.timeout = this.addedToCartToastConfig.addedToCartToast?.timeout;
+    else this.timeout = 3000;
+
+    this.subscription.add(
+      this.cartAddEntrySuccess$
+        .pipe(
+          switchMap((cartEntry) => {
+            return combineLatest([
+              of(cartEntry),
+              this.productService.get(cartEntry.productCode),
+            ]).pipe(
+              filter((result) => result[1] !== undefined),
+              map(([cartEntry, product]) => ({
+                quantityAdded: cartEntry.quantityAdded,
+                product,
+              }))
+            );
+          })
+        )
+        .subscribe(({ quantityAdded, product }) => {
+          let toastContainerClass = `${this.toastContainerBaseClass} `;
+          const toastItem = this.addedToCartToastService.addToast(
+            quantityAdded,
+            product,
+            toastContainerClass
+          );
+          // trigger enter animations with class change
+          setTimeout(() => {
+            toastItem.baseClass = this.getToastStyles();
+          });
+
+          this.scrollEventUnlistener && this.scrollEventUnlistener();
+
+          // wait for enter animation end
+          setTimeout(() => {
+            this.triggerScrollEvent();
+          }, 500);
+
+          // trigger slide out animation after timeout
+          toastItem.timeoutRef = setTimeout(() => {
+            this._closeToast(toastItem);
+          }, this.timeout);
+        })
+    );
+  }
+
+  ngOnDestroy() {
+    this.subscription.unsubscribe();
+  }
+
+  triggerScrollEvent() {
+    this.scrollEventUnlistener = this.renderer.listen(
+      this.winRef.nativeWindow,
+      'scroll',
+      this._setPosition
+    );
+  }
+
+  isHeaderInView(): boolean {
+    if (!this.headerElement) return false;
+    const headerBounding = this.headerElement.getBoundingClientRect();
+    return headerBounding.bottom >= 0;
+  }
+
+  isHeaderFixed() {
+    if (!this.headerElement) return false;
+    const headerPosition = this.headerElement.style.position;
+    return headerPosition === 'fixed';
+  }
+
+  closeToast(toastItem: CartToastItem) {
+    toastItem.timeoutRef && clearTimeout(toastItem.timeoutRef);
+    this._closeToast(toastItem);
+  }
+
+  _closeToast = (toastItem: CartToastItem) => {
```

Comment:
why _ here?

Reply:
I'm trying to keep this private and it has the same name as another method. Let me know if I should change the name.